src/hooks/transactor/PayProjectTx.ts

```diff
@@ -13,20 +13,27 @@ export function usePayProjectTx(): TransactorInstance<{
 }> {
   const { transactor, contracts } = useContext(UserContext)
   const { terminal, projectId } = useContext(ProjectContext)
-  const { userAddress } = useContext(NetworkContext)
+  const { userAddress, onSelectWallet } = useContext(NetworkContext)
+
+  const checkWalletConnected = () => {
```

Comment:
@johnnyd-eth do you think we could just add this flow into the useTransactor hook itself? since any transaction in the app will need to have a connected wallet

Reply:
> @johnnyd-eth do you think we could just add this flow into the useTransactor hook itself? since any transaction in the app will need to have a connected wallet

Yep good idea, I'll do that

Reply:
@peripheralist Moved it into the useTransactor hook 👍